solutions/gold/usaco-231

```diff
@@ -0,0 +1,153 @@
+---
+id: usaco-231
+source: USACO Gold 2013 January
+title: Seating
+author: Chongtian Ma
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_seating.html)
+
+## Implementation
+
+Let's denote a spot as filled if its value is $0$ and empty if its value is $1$.
+The problem boils down to, for each A query find the leftmost consecutive
+segments of 1s and fill it with 0s, and for each L query set all values in that
+range to 1. This can be maintained with a lazy propagated segment tree.
+
+For each segment in our tree, as the official editorial suggests, we need to
+maintain three pieces of information: $\mathtt{seg}$ as the maximum amount of
+consecutive $1$s, $\mathtt{pref}$ as the amount of $1$s from the left of the
+segment, and $\mathtt{suff}$ as the amount of $1$s from the right of the
+segment. When merging two segments, make sure to account for overlapping
```

Comment:
```suggestion
segment. When merging two segments, make sure to account for overlapping
```

Explain that the largest interval can either lie in either half or between them, which is why we need to maintain "pref" and "suff" in the first place.